respect-datalayer/src/commonMain/kotlin/world/respect/datalayer/school/PersonQrDataSource.kt

```diff
@@ -0,0 +1,49 @@
+package world.respect.datalayer.school
+
+import io.ktor.http.Parameters
+import kotlinx.coroutines.flow.Flow
+import world.respect.datalayer.DataLoadParams
+import world.respect.datalayer.DataLoadState
+import world.respect.datalayer.school.model.PersonBadge
+import world.respect.datalayer.shared.WritableDataSource
+import world.respect.datalayer.shared.params.GetListCommonParams
+
+interface PersonQrDataSource : WritableDataSource<PersonBadge> {
+    data class GetListParams(
+        val common: GetListCommonParams = GetListCommonParams(),
+    ) {
+        companion object {
+            fun fromParams(params: Parameters): GetListParams {
+                return GetListParams(
+                    common = GetListCommonParams.fromParams(params)
+                )
+            }
+        }
+    }
+
+    suspend fun listAll(
+        listParams: GetListParams = GetListParams(),
+    ): DataLoadState<List<PersonBadge>>
+
+    fun listAllAsFlow(
+        loadParams: DataLoadParams = DataLoadParams(),
+        listParams: GetListParams = GetListParams(),
+    ): Flow<DataLoadState<List<PersonBadge>>>
+
+    fun findByGuidAsFlow(guid: String): Flow<DataLoadState<PersonBadge>>
+
+    suspend fun deletePersonBadge(
```

Comment:
This delete function must be removed (as mentioned above because its a synchronized system that runs offline we need to use the status enum). Use the store function/status enum.